packages/datadog-plugin-kafkajs/src/producer.js

```diff
@@ -1,13 +1,22 @@
 'use strict'
 
 const ProducerPlugin = require('../../dd-trace/src/plugins/producer')
+const { encodePathwayContext } = require('../../dd-trace/src/datastreams/pathway')
 
 class KafkajsProducerPlugin extends ProducerPlugin {
   static get id () { return 'kafkajs' }
   static get operation () { return 'produce' }
 
   start ({ topic, messages }) {
-    const span = this.startSpan({
+
+    let pathwayCtx
+    if (this.config.dsmEnabled) {
```

Comment:
This block could be extracted to the base class as a helper.

Reply:
I think we'll do this when we support more queues